src/components/modals/FundAccountModal.tsx

```diff
@@ -0,0 +1,117 @@
+import { WalletListModal, WalletListResult } from './WalletListModal'
+
+interface Props {
+  bridge: AirshipBridge<void>
+  navigation: NavigationBase
+}
+
+/**
+ * Renders a modal with descritive options for funding an account
+ */
+export const FundAccountModal = (props: Props) => {
+  const { bridge, navigation } = props
+  const theme = useTheme()
+  const style = styles(theme)
+  const dispatch = useDispatch()
+
+  const activeUsername = useSelector(state => state.core.account.username)
+  const isLightAccount = activeUsername == null
+
+  const handleBuy = useHandler(() => {
+    if (isLightAccount) {
+      showBackupForTransferModal(() => navigation.navigate('upgradeUsername', {}))
+    } else {
+      navigation.navigate('buyTab', { screen: 'pluginListBuy' })
+    }
+    Airship.clear()
+  })
+
+  const handleReceive = useHandler(async () => {
+    Airship.clear()
+    if (isLightAccount) {
+      showBackupForTransferModal(() => navigation.navigate('upgradeUsername', {}))
+    } else {
+      const result = await Airship.show<WalletListResult>(bridge => (
+        <WalletListModal bridge={bridge} headerTitle={lstrings.select_receive_asset} navigation={navigation} showCreateWallet />
+      ))
+
+      if (result?.type === 'wallet') {
+        const { walletId, tokenId } = result
+        await dispatch(selectWalletToken({ navigation, walletId, tokenId }))
+        navigation.navigate('request', { tokenId, walletId })
+
+        await Airship.show(bridge2 => (
+          <ModalUi4 bridge={bridge2} title={lstrings.fund_account_modal_receive_title} onCancel={() => bridge2.resolve()}>
+            <ModalMessage>{lstrings.fund_account_modal_receive_message}</ModalMessage>
+          </ModalUi4>
+        ))
+      }
+    }
+  })
+
+  const handleCancel = useHandler(() => {
+    bridge.resolve()
+  })
+
+  const iconProps = React.useMemo(() => ({ size: theme.rem(1.25), color: theme.iconTappable }), [theme])
+
+  return (
+    <ModalUi4 bridge={bridge} title={lstrings.fund_account_modal_title} onCancel={handleCancel}>
+      <SelectableRow
+        marginRem={0.5}
+        minimumFontScale={0.5}
+        title={lstrings.buy_crypto}
+        subTitle={lstrings.fund_account_modal_buy_body}
+        onPress={handleBuy}
+        icon={
+          <View style={style.iconContainer}>
+            <FontAwesomeIcon name="bank" {...iconProps} />
+            <AntDesignIcon name="arrowright" {...iconProps} />
+          </View>
+        }
+      />
+      <SelectableRow
+        marginRem={0.5}
+        minimumFontScale={0.5}
+        title={lstrings.fund_account_modal_receive_title}
+        subTitle={sprintf(lstrings.fund_account_modal_receive_body_1s, config.appName)}
+        onPress={handleReceive}
+        icon={
+          <View style={style.iconContainer}>
+            <AntDesignIcon name="wallet" {...iconProps} />
+            <AntDesignIcon name="arrowright" {...iconProps} />
+          </View>
+        }
+      />
+    </ModalUi4>
+  )
```

Comment:
This really should have an OK button to close the modal. Looks weird with no button.